docker/kubernetes-tentacle/bootstrapRunner/bootstrapRunner.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+)
+
+// The bootstrapRunner applet is designed to execute a script in a specific folder
+// and format the script's output from stdout and stderr into the following format:
+// <line number>|<RFC3339Nano date time>|<"stdout" or "stderr">|<original string>
+//
+// The applet can be executed as such:
+// bootstrapRunner <working directory> <script> <script arguments>...
+//
+// Note: all arguments given after the <script> argument are passed directly to the script as arguments.
+func main() {
+	workspacePath := os.Args[1]
+	args := os.Args[2:]
+	cmd := exec.Command("bash", args[0:]...)
+	cmd.Dir = workspacePath
+	stdOutCmdReader, _ := cmd.StdoutPipe()
+	stdErrCmdReader, _ := cmd.StderrPipe()
+
+	stdOutScanner := bufio.NewScanner(stdOutCmdReader)
+	stdErrScanner := bufio.NewScanner(stdErrCmdReader)
+
+	doneStd := make(chan bool)
+	doneErr := make(chan bool)
+
+	//stdout log file
+	so, err := os.Create(workspacePath + "/stdout.log")
+	if err != nil {
+		panic(err)
+	}
+	// close fo on exit and check for its returned error
+	defer func() {
+		if err := so.Close(); err != nil {
+			panic(err)
+		}
+	}()
+	// make a write buffer
+	stdoutLogFile := bufio.NewWriter(so)
+
+	//stderr log file
+	se, err := os.Create(workspacePath + "/stderr.log")
+	if err != nil {
+		panic(err)
+	}
+	// close fo on exit and check for its returned error
+	defer func() {
+		if err := se.Close(); err != nil {
+			panic(err)
+		}
+	}()
+	// make a write buffer
+	stderrLogFile := bufio.NewWriter(se)
+
+	go reader(stdOutScanner, stdoutLogFile, &doneStd)
+	go reader(stdErrScanner, stderrLogFile, &doneErr)
+
+	err = cmd.Start()
+	if err != nil {
+		panic(err)
+	}
+
+	// Wait for output buffering first
+	<-doneStd
+	<-doneErr
+
+	err = cmd.Wait()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error waiting for Cmd", err)
+		os.Exit(1)
+	}
+}
+
+func reader(scanner *bufio.Scanner, writer *bufio.Writer, done *chan bool) {
+	for scanner.Scan() {
+		message := fmt.Sprintf("%s|%s\n", time.Now().UTC().Format(time.RFC3339Nano), scanner.Text())
```

Comment:
I don't love that this doesn't have the UTC timezone in the timestamp (ISO 8601). We'll just need to make sure the Tentacle code parses these and knows they are UTC

Reply:
Good point, I'll have a fiddle with this to see if I can get ISO8601 exactly.

Reply:
Ok, apparently `RFC3339` is for all practical applications, ISO 8601 compliant: https://stackoverflow.com/q/522251

I'm getting `2024-03-19T01:28:24.226953Z` output, where the `Z` represents UTC I believe and I think this is compliant with ISO 8601.

Reply:
ok yep, that's good then 👍